src/main/java/org/jenkinsci/plugins/xunit/XUnitChecksPublisher.java

```diff
@@ -0,0 +1,180 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.jenkinsci.plugins.xunit;
+
+import edu.hm.hafner.util.VisibleForTesting;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.tasks.junit.CaseResult;
+import hudson.tasks.junit.TestResult;
+import hudson.tasks.junit.TestResultSummary;
+import io.jenkins.plugins.checks.api.ChecksConclusion;
+import io.jenkins.plugins.checks.api.ChecksDetails;
+import io.jenkins.plugins.checks.api.ChecksOutput;
+import io.jenkins.plugins.checks.api.ChecksPublisher;
+import io.jenkins.plugins.checks.api.ChecksPublisherFactory;
+import io.jenkins.plugins.checks.api.ChecksStatus;
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.plugins.displayurlapi.DisplayURLProvider;
+
+import java.util.List;
+
+/**
+ * Publishes test status to Checks API
+ * Implementation is a copy of:
+ * https://github.com/jenkinsci/junit-plugin/blob/dd65de93befc357d7160251536c903914dd257c6/src/main/java/io/jenkins/plugins/junit/checks/JUnitChecksPublisher.java
+ */
+class XUnitChecksPublisher {
+    public static final String SEPARATOR = ", ";
+
+    // cap to avoid hitting check API message limit
+    private static final int MAX_MSG_SIZE_TO_CHECKS_API = 65535;
+    private final Run<?, ?> run;
+    private final String checksName;
+    private final TestResult result;
+    private final TestResultSummary summary;
+
+    public XUnitChecksPublisher(final Run<?, ?> run, final String checksName, final TestResult result, final TestResultSummary summary) {
+        this.run = run;
+        this.checksName = checksName;
+        this.result = result;
+        this.summary = summary;
+    }
+
+    public void publishChecks(TaskListener listener) {
+        ChecksPublisher publisher = ChecksPublisherFactory.fromRun(run, listener);
+        publisher.publish(extractChecksDetails());
+    }
+
+    @VisibleForTesting
+    ChecksDetails extractChecksDetails() {
+        String testsURL = DisplayURLProvider.get().getTestsURL(run);
+        ChecksOutput output = new ChecksOutput.ChecksOutputBuilder()
+                .withTitle(extractChecksTitle())
+                .withSummary("<sub>Send us [feedback](https://github.com/jenkinsci/xunit-plugin/issues)")
+                .withText(extractChecksText(testsURL))
+                .build();
+
+        return new ChecksDetails.ChecksDetailsBuilder()
+                .withName(checksName)
+                .withStatus(ChecksStatus.COMPLETED)
+                .withConclusion(summary.getFailCount() > 0 ? ChecksConclusion.FAILURE : ChecksConclusion.SUCCESS)
```

Comment:
I'm curious if i should consider the configured `threshold`'s here so only when one of the `threshold`'s is exceeded the result is `ChecksConclusion.FAILURE`?

Reply:
yes, good catch, I miss this check. Failure or success is always subject to the thresholds